src/lualib/ArrayEntries.ts

```diff
@@ -0,0 +1,14 @@
+// https://262.ecma-international.org/10.0/#sec-array.prototype.entries
+function __TS__ArrayEntries<T>(this: void, array: T[]): IterableIterator<[number, T]> {
+    let key = 0;
+    return {
+        [Symbol.iterator](): IterableIterator<[number, T]> {
+            return this;
+        },
+        next(): IteratorResult<[number, T]> {
+            const result = { done: array[key] === undefined, value: [key, array[key]] as [number, T] };
+            key++;
+            return result;
+        },
+    };
+}
```

Comment:
Why not just this?
```ts
function __TS__ArrayEntries<T>(this: void, array: T[]) {
    return ipairs(array);
}
```

Reply:
Because TS defines the type of entries as returning `IterableIterator<[number, T]>`, and I don't think ipairs `LuaIterable<LuaMultiReturn<[number, NonNullable<T>]>>` is 1:1 assignable to that since it's a different iterator kind and also the lack of LuaMultiReturn.

Reply:
Ah yeah, that would be a problem if used indirectly. It's a shame since this version is so much more complex.

It might be good to add a test for indirect use:
```
...
const entries = array.entries();
for (const [i, v] of entries) {
...
```

Reply:
Good call, added a test for indirect use as well as destructuring, those both fail if ipairs is used